test/extensions/filters/http/rate_limit_quota/config_test.cc

```diff
@@ -63,6 +64,94 @@ TEST(RateLimitQuotaFilterConfigTest, RateLimitQuotaFilterWithCorrectProto) {
   Http::FilterFactoryCb cb = factory.createFilterFactoryFromProtoTyped(
       filter_config, stats_prefix, mock_stream_client->context_);
   cb(filter_callback);
+
+  GlobalTlsStores::clear();
+}
+
+TEST(RateLimitQuotaFilterConfigTest, RateLimitQuotaFilterWithInvalidMatcher) {
+  std::string filter_config_yaml = R"EOF(
+  rlqs_server:
+    envoy_grpc:
+      cluster_name: "rate_limit_quota_server"
+  domain: test
+  bucket_matchers:
+    matcher_list:
+      matchers:
+        # Assign requests with header['env'] set to 'staging' to the bucket { name: 'staging' }
```

Comment:
This comment doesn't seem relevant for this case - should it be updated to indicate the filter contains an invalid matcher (because of the type mismatch)?